File: Valid-Palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(text) {
    let i = 0;
    let j = text.length - 1;

    while (i < j) {
        while (i < j && !/[a-zA-Z0-9]/.test(text[i])) i++;
        while (i < j && !/[a-zA-Z0-9]/.test(text[j])) j--;

        if (text[i].toLowerCase() !== text[j].toLowerCase()) return false;

        i++;
        j--;
    }

    return true;
};

File: Valid-Palindrome.java
public class Solution {
    public boolean isPalindrome(String text) {
        int i = 0;
        int j = text.length() - 1;

        while (i < j) {
            while (i < j && !Character.isLetterOrDigit(text.charAt(i))) i++;
            while (i < j && !Character.isLetterOrDigit(text.charAt(j))) j--;

            if (Character.toLowerCase(text.charAt(i)) != Character.toLowerCase(text.charAt(j))) {
                return false;
            }

            i++;
            j--;
        }

        return true;
    }
}
